### PR TITLE
Ft update platforms

### PR DESCRIPTION
**Pull request by esirK**
## Description
Performs
1. Python 2 to Python3 upgrade
2. requirements.txt packages update to the latest versions
3. Add dockerization
4. Add CI/CD (in progress)
5. Update Readme (in progress)

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Chore (non-breaking change which does not add visible functionality but improves code quality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,8 @@
+MAIL_PASSWORD=
+MAIL_RECIPIENTS=
+MAIL_SERVER=smtp.sendgrid.net
+MAIL_USERNAME=apikey
+SECRT_KEY=dsadsadsad
```
Comment (kilemensi) [inline .env.template]:
```suggestion
SECRET_KEY=Bk12miuhnZfv4pOLf3SyPtgI7ahrgUIRgbjK6IPRZQs=
```

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Move cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+      - name: Run app
+        run: |
+          echo "TWOOPSTRACKER_SECRET_KEY=${{ secrets.TWOOPSTRACKER_SECRET_KEY }}" >> .env
+          echo "TWOOPSTRACKER_SENTRY_DSN=${{ secrets.TWOOPSTRACKER_SENTRY_DSN }}" >> .env
+          echo "TWOOPSTRACKER_GOOGLE_OAUTH2_CLIENT_SECRET=${{ secrets.TWOOPSTRACKER_GOOGLE_OAUTH2_CLIENT_SECRET }}" >> .env
+          echo "TWOOPSTRACKER_GOOGLE_OAUTH2_CLIENT_ID=${{ secrets.TWOOPSTRACKER_GOOGLE_OAUTH2_CLIENT_ID }}" >> .env
```
Comment (kilemensi) [inline .github/workflows/pull_request.yml]: `TWOOPSTRACKER_`?

File: app.py [hunk 3]
```diff
@@ -1,12 +1,9 @@
-from flask_script import Manager
-from flask_migrate import Migrate, MigrateCommand
+from flask_migrate import Migrate
 
 from healthtools_ec.core import app
 from healthtools_ec.models import db
 
 migrate = Migrate(app, db)
```
Comment (kilemensi) [inline app.py:6]: Is this (`migrate`) being used anywhere?

File: healthtools_ec/initiates.py [hunk 4]
```diff
@@ -68,28 +71,28 @@ def initiates_home_mobi():
             db.session.commit()
             response = email_initiates(initiate)
             print(response)
-            if session['lang']:
-                return render_template('mobile/initiates/initiateredirect_xh.html')
+            if session["lang"]:
+                return render_template("mobile/initiates/initiateredirect_xh.html")
```
Comment (kilemensi) [inline healthtools_ec/initiates.py]: Something like
```python
    # Default is English so we don't need locale extension
    template_locale = ""
    if session["lang"]:
        template_locale = "_xh"
```

Then we only need  a single `render_template`:
```python
    render_template(f"mobile/initiates/initiateredirect{template_locale}.html")
```

**Review by kilemensi — COMMENTED** [review]
👍🏽 


----


Still in the call but looks and runs fine on my end.

File: healthtools_ec/initiates.py [hunk 5]
```diff
@@ -25,41 +32,37 @@ def initiates_home():
...
-                else:
-                    flash('Please correct the problems below and try again.', 'warning')
-
-    if not request.is_xhr:
-        if session['lang']:
-            resp = make_response(render_template('initiates/initiates_xh.html', form=form))
-        else:
-            resp = make_response(render_template('initiates/initiates.html', form=form))
-
+                flash("Please correct the problems below and try again.", "warning")
```
Comment (KhadijaMahanga) [inline healthtools_ec/initiates.py]: Aren't we doing the same thing on `if` and `else`

File: healthtools_ec/locate.py [hunk 6]
```diff
@@ -1,91 +1,91 @@
...
-                    flash('Please correct the problems below and try again.', 'warning')
-                else:
-                    flash('Please correct the problems below and try again.', 'warning')
-
-    if not request.is_xhr:
-        if session['lang']:
-            resp = make_response(render_template('mobile/find/find_xh.html', form=form))
-        else:
-            resp = make_response(render_template('mobile/find/find.html', form=form))
+                flash("Please correct the problems below and try again.", "warning")
```
Comment (KhadijaMahanga) [inline healthtools_ec/locate.py]: and here too

**Review by KhadijaMahanga — COMMENTED** [review]
Looks Good

File: healthtools_ec/initiates.py (hunk 5 above)
Comment (esirK) [reply healthtools_ec/initiates.py]: We are using different `template` depending on the `lang` but I'm updating this to what @kilemensi  suggested [here](https://github.com/CodeForAfrica/HealthTools.ZA-Ezolwaluko/pull/2/files/fb9a92d0db03f5f166430bc7ceb617782e41db5d#r843694206)

Comment (KhadijaMahanga) [reply healthtools_ec/initiates.py]: @esirK I speaking about line `40, 41, 42 and 43`. Else of form.validate()

it reads from 
```
if form.validate():
   ...
   ...
else:
   if session["lang"]:
      flash("Please correct the problems below and try again.", "warning")
   else:
      flash("Please correct the problems below and try again.", "warning")
```

Comment (esirK) [reply healthtools_ec/initiates.py]: Oooh, I see. I think the intent is to have an `XHOSA` version of the response. I'll look for a translation although I think adding a localization might be better.

File: app.py (hunk 3 above)
Comment (esirK) [reply app.py:6]: yes it's used in different places e.g [here](https://github.com/CodeForAfrica/HealthTools.ZA-Ezolwaluko/blob/ft-update-platforms/migrations/env.py#L24)

**Pull request merged**